locals.tf

```diff
@@ -1,6 +1,8 @@
 locals {
-  is_linux                                   = var.image_os == "linux"
-  is_windows                                 = var.image_os == "windows"
-  network_interface_ip_configuration_indexes = var.new_network_interface == null ? [] : toset(range(length(var.new_network_interface.ip_configurations)))
-  patch_mode                                 = coalesce(var.patch_mode, local.is_linux ? "ImageDefault" : "AutomaticByOS")
+  is_linux                                               = var.image_os == "linux"
+  is_windows                                             = var.image_os == "windows"
+  network_interface_ip_configuration_indexes             = var.new_network_interface == null ? [] : toset(range(length(var.new_network_interface.ip_configurations)))
+  patch_mode                                             = coalesce(var.patch_mode, local.is_linux ? "ImageDefault" : "AutomaticByOS")
+  bypass_platform_safety_checks_on_user_schedule_enabled = local.patch_mode == "AutomaticByPlatform" ? var.bypass_platform_safety_checks_on_user_schedule_enabled : false
```

Comment:
I'd like to use the variables directly instead of adding these `local`. Could we add `precondition` check on the resource blocks to throw an error instead of setting values for the users? I don't want to surprise them when `patch_mode` is not `AutomaticByPlatform` and `var.bypass_platform_safety_checks_on_user_schedule_enabled` is `true`.